src/testing/time_refinement_study_reference.cpp

```diff
@@ -20,24 +20,35 @@ TimeRefinementStudyReference<dim, nstate>::TimeRefinementStudyReference(
 template <int dim, int nstate>
 Parameters::AllParameters TimeRefinementStudyReference<dim,nstate>::reinit_params_for_reference_solution(int number_of_timesteps, double final_time) const
 {
-     PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
-
-     const double dt = final_time/number_of_timesteps;     
-     parameters.ode_solver_param.initial_time_step = dt;
-     
-     pcout << "Using timestep size dt = " << dt << " for reference solution." << std::endl;
-     
-     return parameters;
+    PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
+
+    const double dt = final_time/number_of_timesteps;     
+    parameters.ode_solver_param.initial_time_step = dt;
+    parameters.flow_solver_param.final_time = final_time;
+
+    //Change to RK because at small dt RRK is more costly but doesn't impact solution much
+    using ODESolverEnum = Parameters::ODESolverParam::ODESolverEnum;
+    parameters.ode_solver_param.ode_solver_type = ODESolverEnum::runge_kutta_solver;
+
+    pcout << "Using timestep size dt = " << dt << " for reference solution." << std::endl;
+
+    return parameters;
 }
 
 template <int dim, int nstate>
 Parameters::AllParameters TimeRefinementStudyReference<dim,nstate>::reinit_params_and_refine_timestep(int refinement) const
 {
-     PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
-     
-     parameters.ode_solver_param.initial_time_step *= pow(refine_ratio,refinement);
-     
-     return parameters;
+    PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
+
+    parameters.ode_solver_param.initial_time_step *= pow(refine_ratio,refinement);
+
+    //For RRK, do not end at exact time because of how relaxation parameter convergence is calculatd
+    using ODESolverEnum = Parameters::ODESolverParam::ODESolverEnum;
+    if (parameters.ode_solver_param.ode_solver_type == ODESolverEnum::rrk_explicit_solver){
+        parameters.flow_solver_param.end_exactly_at_final_time = false;
```

Comment:
Setting `end_exactly_at_final_time` directly in the .prm file might be better unless several ode_solver_types are being run with the same .prm file.

Reply:
I would rather leave it in the test out of personal preference. I tend to test using the same .prm file and would rather change only one line (`ode_solver_type`) than two.

Reply:
Sure, sounds good.